### PR TITLE
Refactor stubs for `marked.js`.

### PR DESCRIPTION
**Pull request by Lalit3716**
- I have moved the declaration file for `marked.js` to it's correct place i.e inside `third` directory.
- Used correct syntax in the declaration file to export types.
- Modified some incorrect type definitions.

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

### CODE REVIEW

File: web/src/markdown.ts [hunk 1]
```diff
@@ -690,12 +688,13 @@ export function parse({
 
     // No <code> around our code blocks instead a codehilite <div> and disable
     // class-specific highlighting.
-    renderer.code = (code: string): string => fenced_code.wrap_code(code) + "\n\n";
+    renderer.code = (code) => fenced_code.wrap_code(code) + "\n\n";
```
Comment (evykassirer) [inline web/src/markdown.ts]: Why did you remove these type annotations?

**Comment by evykassirer** [comment]
I've integrated most of your feedback, very helpful thanks!

File: web/src/markdown.ts (hunk 1 above)
Comment (Lalit3716) [reply web/src/markdown.ts]: Because we don’t need to be explicit here, typescript is able to infer types for us here.
But if you think it’s nice to be explicit then sure.
